concepts/Vulnerability_disclosure_and_embargo_policy/00004-Vulnerability_disclosure_and_embargo_policy.md

```diff
@@ -0,0 +1,221 @@
+# RFC00004: Vulnerability Disclosure and Embargo Policy
+- Authors: [axel simon](github@axelsimon.net)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2020-03-10
+- Status Note: under discussion
+- Supersedes: N/A
+- Start Date: 2020-03-03 (date you started working on this idea)
+- Tags: security, infrastructure
+
+## Summary
+
+Describe the security vulnerability disclosure process, including embargoes.
+
+## Motivation
+
+Users and researchers who discover security issues in Enarx need to know that
+we welcome their efforts and how we will manage their disclosure. They must
+also be able to have an idea of how we use security embargoes.
+
+In other words:
+- We want:
+  - to encourage reporting of security issues (ie: “attack us!”)
+  - to make it easy to report security issues
+  - to maintain and extend our policy of safe defaults / discouraging people
+  from adopting bad security practices
+  - to recognise and show appreciation for the time spent researching and
+  reporting
+  - to make it possible for people who may be at risk to report issues
+  pseudonymously
+
+- We need:
+  - a clear policy, describing both:
+    - how to report; and
+    - what to expect during the process
+  - a main reporting channel:
+    - secondary channels, should the main one fail/not be available/be
+  overloaded.
+  - a list of people who can be reached directly
+  - a place to list security reports and give attribution
+
+
+## Tutorial
+
+This RFC aims to provide a set of documents (wiki pages) that describe the 
+Enarx vulnerability disclosure policy and the list of security advisories and
+ fixes.
+
+The two pages are "Vulnerability Reporting and Embargo Policy" and "Enarx 
+Security Advisories"
+
+--- 
+### Vulnerability Reporting and Embargo Policy
+The Enarx project aims to be as safe to use as possible and therefore to 
+produce code that is as secure as possible. As such, we welcome any and all 
+research into Enarx and appreciate your efforts to take the time to 
+responsibly disclose any issues you find.
+
+Vulnerabilities may be discovered in:
+1. Enarx code
+2. Enarx dependencies (including compilers)
+3. Enarx documentation
+4. 3rd party hardware or firmware
+
+All security bugs in Enarx should be reported by email to security@enarx.io. 
+This list is delivered to a small security team. Your email will be 
+acknowledged within 48 hours, and you’ll receive a more detailed response to 
+your email within 96 hours indicating the next steps in handling your report.
+ If you wish, you may encrypt your report using our public key. This key is 
```

Comment:
I have never been on a security council. Should we _require_ an encrypted email for receiving reports? I'm also not savvy about email security in general...

Reply:
I have also never been on a security council, but having looked at security reporting methods for other companies, it's very much the norm to require a confidential communication channel.

Personally, I'm in favor of enforcing encrypted comms (even if it doesn't end up being encrypted email) as a requirement for vulnerability disclosure, if only as an indication that we take this process seriously. If we don't bother to secure your vulnerability disclosure, what does that say about our project's attitude toward security in general? Not good things, I posit.

Reply:
This is the crux of the document i think, as has been mentioned elsewhere in the conversations about this RFC: do we go through the usual email reporting method, or do we skip email altogether / treat it as a public channel, and require reporters to *not* use it for anything confidential (cf. [this analysis](https://latacora.micro.blog/2019/07/16/the-pgp-problem.html) if you'd like more details @connorkuehl).
Further to your point Connor, you can't *require* encrypted email. To my knowledge, there is no mechanism to refuse unencrypted email. Even if the recipient's server is configured to refuse any clear text email, the sender's one isn't, and any hops in between either. Email is simply built for a simpler time on the Internet were you didn't need to assume anyone was really malicious.

I entirely agree with your point @mbestavros.